### PR TITLE
Graceful shutdown relay chain

### LINKED ISSUE #164: The relay chain doesn't have a clean shutdown
~~When ran without --validator, a full node is started,~~ _For the relay chain,_ a new task manager is created but `graceful_shutdown` is never called on that task manager.

We could implement something in substrate that would allow adding a sub-task manager to a task manager and calls `graceful_shutdown` on every sub-task manager.

_[Update: this issue also exists for --validator]_

### PR DESCRIPTION
**Pull request by cecton**
Fixes #164

### CODE REVIEW

File: test/parachain/src/service.rs [hunk 1]
```diff
@@ -191,21 +190,17 @@ pub fn run_collator(
...
-		// Make sure the polkadot task manager survives as long as the service.
-		let polkadot_future = polkadot_future.then(move |_| {
-			let _ = polkadpt_task_manager;
-			ready(())
-		});
-
 		task_manager
 			.spawn_essential_handle()
-			.spawn("polkadot", polkadot_future);
+			.spawn("polkadot-collator", polkadot_future);
```
Comment (bkchr) [inline test/parachain/src/service.rs:198]:
```suggestion
			.spawn("polkadot", polkadot_future);
```

This is the future of the Polkadot service.

**Comment by bkchr** [comment]
bot merge

File: test/parachain/src/service.rs (hunk 1 above)
Comment (cecton) [reply test/parachain/src/service.rs]: hmmm no I think it's the collator future: https://github.com/paritytech/polkadot/blob/master/collator/src/lib.rs#L263

It's a big loop that runs "collation work" when there are new notifications.

What I would call the "polkadot service" is the task manager that is passed as child right after this.

Comment (cecton) [reply test/parachain/src/service.rs]: I guess my vocabulary is still a big wrong :sweat:

**Pull request merged**